lib/browser-runtime.ts

```diff
@@ -423,6 +434,11 @@ function createPickle(context: CompositionContext, pickle: messages.Pickle) {
   const suiteOptions = tags
     .filter(looksLikeOptions)
     .map(tagToCypressOptions)
+    .filter((tag) =>
```

Comment:
Instead of silently filtering this out, I think it should error similarly to how Cypress does (see your example in run-mode).

Reply:
I had a look at this but each pickle has a list of tags, following cucumber's inheritance rules. There's no reference back to the node(s) the tag is on. Here I can't tell if `@testIsolation(<value>)` is on the current scenario or a parent `Rule` or `Feature`.

I could traverse the document and build a mapping of node to tag but that won't work as scenario names aren't required to be unique.

Reply:
A pickle has a `astNodeIds` property, which contains a reference to the scenario-node, and you can use `context.astIdsMap` to obtain this. This is an example of this in `lib/browser-runtime.ts` already.

PS: I suspect that `astNodeIds` contains more reference in cases of scenario _outlines_, so make sure to check that out and grab the correct one (and test coverage).

Reply:
Tips: you can use `gherkin-javascript` and [jq](https://jqlang.github.io/jq/) to inspect the protocol.

```
$ npx gherkin-javascript --no-source tmp/features/ambiguous_keywords.feature_3/cypress/e2e/a.feature | jq .

<JSON output>
```

Reply:
a6746ed9790d49234ee8a86ef8f173f745caaf07

`gherkin-javascript` was very useful -- thanks!

I couldn't use the `astIdsMap` as a AST nodes have no parent reference so I couldn't go from tag node to the parent node. Instead I traverse the document and if the tag is defined on the current scenario node then I throw an error.

I could make this more efficient by building a map from tag id to parent node id in one pass.

I still need to look into when `astNodeIds` contains more than one reference -- I wanted to push this and get your thoughts before I do that.

Reply:
> I couldn't use the astIdsMap as a AST nodes have no parent reference so I couldn't go from tag node to the parent node. Instead I traverse the document and if the tag is defined on the current scenario node then I throw an error.

What I meant is that you can grab the pickle's `astNodeIds`, use this to get a hold of the scenario node, which has information about tags that are specific to a single scenario. See below.

```ts
const scenario = assertAndReturn(
  context.astIdsMap.get(
    assertAndReturn(
      pickle.astNodeIds?.[0],
      "Expected to find at least one astNodeId"
    )
  ),
  `Expected to find scenario associated with id = ${pickle.astNodeIds?.[0]}`
);

console.log("Pickle tags", pickle.tags);
console.log("Scenario tags", (scenario as any).tags);
```

For the following tests, this will output a 2-element array and a 1-element array.

```gherkin
@foo
Feature: a feature name
  @bar
  Scenario: a scenario name
    Given a step
```

Reply:
D'oh 🤦 I'm traversing to find the scenario ast node that I can already get from the map.

So, yeah, I can just check if the tag belongs to the corresponding ast node for the scenario -- if so throw an error. Otherwise carry on and just filter it out as it was defined on a parent Feature or Rule.

Reply:
Done 5cde757

Not sure how I got myself so tied in knots with this one 😅 .